core/src/CMakeLists.txt

```diff
@@ -1,17 +1,24 @@
 #I have to leave these here for tribits
+IF (KOKKOS_ENABLE_IMPL_DESUL_ATOMICS)
+  find_package(desul_atomics QUIET)
+ENDIF()
+
 KOKKOS_INCLUDE_DIRECTORIES(
   ${CMAKE_CURRENT_BINARY_DIR}
   ${CMAKE_CURRENT_SOURCE_DIR}
   ${KOKKOS_TOP_BUILD_DIR}
 )
+IF (KOKKOS_ENABLE_IMPL_DESUL_ATOMICS AND NOT desul_atomics_FOUND)
+  KOKKOS_INCLUDE_DIRECTORIES(
+    ${CMAKE_CURRENT_SOURCE_DIR}/../../tpls/desul/include
```

Comment:
We both use relative path from `CMAKE_CURRENT_SOURCE_DIR` and absolute from `CMAKE_SOURCE_DIR` below.  Is there a reason you can't use `CMAKE_SOURCE_DIR` here?
Also did you check whether `CMAKE_SOURCE_DIR` works with Trilinos?